docs/scos/dev/feature-walkthroughs/202204.0/configurable-product-feature-walkthrough/configurable-product-flow-chart.md

```diff
@@ -0,0 +1,196 @@
+2. A plugin that handles the request is selected based on the configurator type.
+3. The plugin expands the request with additional necessary data:
+  - store, locale, currency, customer, price mode.
+  - `CheckSum`—calculates the CRC32 polynomial of a request data as a string.
+  - `timestamp`—the timestamp when the request is created.
+  - `backUrl`—based on the `Referer` header.
+4. The plugin generates the URL that points to the configurator page together with all the necessary data.
+
+### Phase 4
+
+Redirects the customer to the configurator page using the GET request.
+
+### Phase 5
+
+1. The customer is on the configurator page.
+2. The customer does the configuration.
+3. Submits the configuration form.
+4. The configurator has to redirect to the gateway page with configuration data.
+
+| PARAMETER | VALUE                                                                  | COMMENT |
+|---|------------------------------------------------------------------------|---|
+| `ProductConfigurationInstance.prices` | `{"EUR":{"GROSS_MODE":{"DEFAULT":30000}},{"NET_MODE":{"DEFAULT": 25000}},"priceData":{"volume_prices":[{"quantity": 5,"net_price": 28500,"gross_price": 29000}]}}`                                                              | Sensitive data. |
+| `ProductConfigurationInstance.isComplete` | `1`                                                                    | Sensitive data. |
+| `ProductConfigurationInstance.availableQuantity` | `2`                                                                    |   |
+| `ProductConfigurationInstance.displayData` | Some text of JSON blob—for example, `["color"=>"red", "weight"=> 100]` |   |
+| `ProductConfigurationInstance.configuration` | `"date"=>"23.07.2020", "time"=>"18:45" 4]`                             | Sensitive data. |
+| `ProductConfigurationInstance.timestamp` | `10312313135234`                                                       | Sensitive data, a certain configuration should be valid only a certain amount of the time given. |
+| `sourceType` | `SOURCE_TYPE_PDP`, `SOURCE_TYPE_CART`, `SOURCE_TYPE_WISHLIST_DETAIL`, …                                        |   |
+| `SKU`  | `some_sku`                                                             |   |
+| itemGroupKey | `some_group_key`                                                         |   |
+
+### Phase 6
+
+* After the successful configuration, the customer is redirected to the configurator gateway with a configuration response.
+* The gateway URL does not equal to the back URL; it’s a fixed known URL.
+
+### Phase 7
+
+1. The gateway page receives data.
+2. The data is checked through the execution of the validator plugins stack for received data.
+  - If validation is not successful, the request is redirected to the `backUrl` without saving the of the configuration with a warning message.
+  - If validation is successful, the framework updates the cart item configuration in the cart.
```

Comment:
```suggestion
  - If validation is successful, the cart item configuration is updated in the cart.
```

Reply:
We should prefer active voice over passive voice if possible https://developers.google.com/style/voice?hl=en